### PR TITLE
Continue working on Spring. Add: MovieController, MovieSessionController, AuthenticationController

### PR DESCRIPTION
**Pull request by IrynaDim**

### CODE REVIEW

File: src/main/java/com/dev/cinema/Main.java [hunk 1]
```diff
@@ -89,5 +88,6 @@ public static void main(String[] args) {
                 + shoppingCartService.getByUser(iryna));
         logger.info("Getting all orders by user id: " + iryna.getId()
                 + orderService.getOrderHistory(iryna));
+
```
Comment (DmGrt) [inline src/main/java/com/dev/cinema/Main.java]: Redundant line)

File: src/main/java/com/dev/cinema/config/AppConfig.java [hunk 2]
```diff
@@ -15,7 +15,8 @@
 @ComponentScan(basePackages = {
         "com.dev.cinema.service",
         "com.dev.cinema.dao",
-        "com.dev.cinema.security"
+        "com.dev.cinema.security",
+        "com.dev.cinema.model.dto"
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/config/AppConfig.java]: do we need dto here?

File: src/main/java/com/dev/cinema/controllers/MovieController.java [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+public class MovieController {
+    private final MovieService movieService;
+    private final MovieMapper movieMapper;
+
+    public MovieController(MovieService movieService, MovieMapper movieMapper) {
+        this.movieService = movieService;
+        this.movieMapper = movieMapper;
+    }
+
+    @PostMapping("/add")
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/controllers/MovieController.java]: https://restfulapi.net/resource-naming/

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+import java.util.stream.Collectors;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+@RequestMapping("/session")
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/controllers/MovieSessionController.java]:
```suggestion
@RequestMapping("/movie-session")
```

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    private final MovieSessionService movieSessionService;
+    private final MovieSessionMapper movieSessionMapper;
+
+    public MovieSessionController(MovieSessionService movieSessionService,
+                                  MovieSessionMapper movieSessionMapper) {
+        this.movieSessionService = movieSessionService;
+        this.movieSessionMapper = movieSessionMapper;
+    }
+
+    @PostMapping("/add")
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/controllers/MovieSessionController.java]: https://restfulapi.net/resource-naming/

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+    public void create(@RequestBody MovieSessionRequestDto sessionDto) {
+        movieSessionService.add(movieSessionMapper.convertToEntity(sessionDto));
+    }
+
+    @GetMapping("/{movieId}&{date}")
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/controllers/MovieSessionController.java]:
```suggestion
    @GetMapping("/available")
```

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java [hunk 7]
```diff
@@ -0,0 +1,41 @@
...
+    public List<MovieSessionResponseDto> findAvailableSessions(@PathVariable Long movieId,
+                                                               @PathVariable String date) {
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/controllers/MovieSessionController.java]: what is the difference between PathVariable and RequstParam?

Comment (Sofasmile) [inline src/main/java/com/dev/cinema/controllers/MovieSessionController.java]: let's change String to LocalDate

File: src/main/java/com/dev/cinema/controllers/map/MovieMapper.java [hunk 8]
```diff
@@ -0,0 +1,24 @@
+package com.dev.cinema.controllers.map;
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/controllers/map/MovieMapper.java]:
```suggestion
package com.dev.cinema.controllers.mapper;
```

File: src/main/java/com/dev/cinema/controllers/map/MovieSessionMapper.java [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+        sessionResponseDto.setId(movieSession.getId());
+        sessionResponseDto.setCinemaHallId(movieSession.getCinemaHall().getId());
+        sessionResponseDto.setMovieTitle(movieSession.getMovie().getTitle());
+        sessionResponseDto.setShowTime(movieSession.getShowTime().toString());
+        return sessionResponseDto;
+    }
+
+    public MovieSession convertToEntity(MovieSessionRequestDto requestDto) {
+        MovieSession movieSession = new MovieSession();
+        movieSession.setMovie(movieService.findByTitle(requestDto.getMovieTitle()));
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/controllers/map/MovieSessionMapper.java]:
```suggestion
        movieSession.setMovie(movieService.findById(requestDto.getMovieId()));
```

File: src/main/java/com/dev/cinema/controllers/map/MovieSessionMapper.java [hunk 10]
```diff
@@ -0,0 +1,41 @@
...
+        movieSession.setCinemaHall(cinemaHallService.getById(requestDto.getCinemaHallId()));
+
+        movieSession.setShowTime(LocalDateTime.parse(requestDto.getShowTime(),
+                DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm")));
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/controllers/map/MovieSessionMapper.java]: can we use standard formatter here?

File: src/main/java/com/dev/cinema/controllers/map/MovieSessionMapper.java [hunk 11]
```diff
@@ -0,0 +1,41 @@
...
+        sessionResponseDto.setMovieTitle(movieSession.getMovie().getTitle());
+        sessionResponseDto.setShowTime(movieSession.getShowTime().toString());
+        return sessionResponseDto;
+    }
+
+    public MovieSession convertToEntity(MovieSessionRequestDto requestDto) {
+        MovieSession movieSession = new MovieSession();
+        movieSession.setMovie(movieService.findByTitle(requestDto.getMovieTitle()));
+        movieSession.setCinemaHall(cinemaHallService.getById(requestDto.getCinemaHallId()));
+
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/controllers/map/MovieSessionMapper.java]: remove empty line

File: src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java [hunk 12]
```diff
@@ -0,0 +1,40 @@
+package com.dev.cinema.model.dto;
+
+public class MovieSessionRequestDto {
+    private Long id;
+    private String movieTitle;
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java]:
```suggestion
    private Long movieId;
```

File: src/main/java/com/dev/cinema/model/dto/MovieSessionResponseDto.java [hunk 13]
```diff
@@ -0,0 +1,40 @@
+package com.dev.cinema.model.dto;
+
+public class MovieSessionResponseDto {
+    private Long id;
+    private String movieTitle;
```
Comment (Sofasmile) [inline src/main/java/com/dev/cinema/model/dto/MovieSessionResponseDto.java:6]: let's add movie id too

File: src/main/java/com/dev/cinema/controllers/map/MovieMapper.java (hunk 8 above)
Comment (IrynaDim) [reply src/main/java/com/dev/cinema/controllers/map/MovieMapper.java]: done

File: src/main/java/com/dev/cinema/controllers/map/MovieSessionMapper.java (hunk 11 above)
Comment (IrynaDim) [reply src/main/java/com/dev/cinema/controllers/map/MovieSessionMapper.java]: done

File: src/main/java/com/dev/cinema/model/dto/MovieSessionResponseDto.java (hunk 13 above)
Comment (IrynaDim) [reply src/main/java/com/dev/cinema/model/dto/MovieSessionResponseDto.java:6]: done

File: src/main/java/com/dev/cinema/config/AppConfig.java (hunk 2 above)
Comment (IrynaDim) [reply src/main/java/com/dev/cinema/config/AppConfig.java]: done

File: src/main/java/com/dev/cinema/controllers/MovieController.java (hunk 3 above)
Comment (IrynaDim) [reply src/main/java/com/dev/cinema/controllers/MovieController.java]: done

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java (hunk 4 above)
Comment (IrynaDim) [reply src/main/java/com/dev/cinema/controllers/MovieSessionController.java]: done

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java (hunk 5 above)
Comment (IrynaDim) [reply src/main/java/com/dev/cinema/controllers/MovieSessionController.java]: done

File: src/main/java/com/dev/cinema/controllers/MovieSessionController.java (hunk 7 above)
Comment (IrynaDim) [reply src/main/java/com/dev/cinema/controllers/MovieSessionController.java]: done

Comment (IrynaDim) [reply src/main/java/com/dev/cinema/controllers/MovieSessionController.java]: done

File: src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java (hunk 12 above)
Comment (IrynaDim) [reply src/main/java/com/dev/cinema/model/dto/MovieSessionRequestDto.java]: done

File: src/main/java/com/dev/cinema/dao/CinemaHallDao.java [hunk 14]
```diff
@@ -7,4 +7,6 @@
     CinemaHall add(CinemaHall cinemaHall);
 
     List<CinemaHall> getAll();
+
+    CinemaHall getById(Long id);
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/dev/cinema/dao/CinemaHallDao.java:11]: let' use consistent naming: getById or findById(i would suggest  just get()) and use it everywhere

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**